### PR TITLE
feat(chore): Recommend matching commands

### LINKED ISSUE #905: Suggest matching commands if the user makes a typo
## Describe the Feature

Suggest matching commands if the user mistypes. For instance, `react-native-cli ini` should show up the help message along with a friendly suggestion such as `Did you mean init?`

## Possible Implementations

We can make use of the [didyoumean.js](https://github.com/dcporter/didyoumean.js/) matching engine.

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Summary:
---------
Suggest matching commands if the user makes a typo.
closes #905 

This is how it goes:
----------
Screenshot from 2020-01-13 20-33-23

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
@thymikee it seems `leven` just returns the actual difference between 2 given input strings. What required in the respective case is a check against some group of values which isn't possible. I would say, `didyoumean.js` would be the candidate here. If you're concerned about it's popularity, `Vue-CLI` makes use `didyoumean` for the purpose.

**Comment by thymikee** [comment]
> it seems leven just returns the actual difference between 2 given input strings

Yup, that's all you need, the Levenshtein distance. I'm familiar with `leven` because we use it for Jest CLI and it serves use well. But if Vue is happy with `didyoumean` then sure, why not, it definitely is more specialized in what we need :)

File: packages/cli/package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "connect": "^3.6.5",
     "cosmiconfig": "^5.1.0",
     "deepmerge": "^3.2.0",
+    "didyoumean": "^1.2.1",
```
Comment (thymikee) [inline packages/cli/package.json]: Did you run `yarn` so that `yarn.lock` is updated?

Comment (jamesgeorge007) [reply packages/cli/package.json]: I use `npm` and always miss it out!

**Comment by thymikee** [comment]
CI is still failing, please make sure the linter passes.

**Comment by jamesgeorge007** [comment]
\cc @thymikee

File: packages/cli/src/index.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import chalk from 'chalk';
 import childProcess from 'child_process';
 import commander from 'commander';
+import didYouMean from 'didyoumean';
```
Comment (Taym95) [inline packages/cli/src/index.ts]: eslint giving this error `Unable to resolve path to module 'didyoumean'  import/no-unresolved`!

Comment (grabbou) [reply packages/cli/src/index.ts]: Linter passing upstream, so going to ignore this and ship it. Will be fixed later if needed, thank you!

**Comment by grabbou** [comment]
Great feature, thank you for working on it and @thymikee for spending time on reviewing it! Let me merge it!

**Pull request merged**